mission_control/src/behaviors/waypoint.cpp

```diff
@@ -179,128 +222,41 @@ void WaypointBehavior::latLongtoUTM(double latitude, double longitude, double* p
 
-  msg.depth = m_depth;
-  msg.latitude = m_lat;
-  msg.longitude = m_long;
-  msg.speed_knots = m_speed_knots;
+    // formula for distance between two 3D points is d=sqrt((x2-x1)^2 + (y2-y1)^2 + (z2-z1)^2)
+    // investigating hypot it looks like it does the squaring and square root to figure out the
+    // return value. Need to add depth or altitude (whatever we are using).
+    //   dist_to_wp = std::hypot((currentNorthing-desiredNorthing, currentEasting-desiredEasting,
+    //   data.depth-m_depth);
+    // dist_to_wp = hypot(abs(currentNorthing-desiredNorthing), abs(currentEasting-desiredEasting));
 
-  msg.ena_mask = 0x0;
-  if (m_depth_ena) msg.ena_mask |= Waypoint::DEPTH_ENA;
-  if (m_lat_ena) msg.ena_mask |= Waypoint::LAT_ENA;
-  if (m_long_ena) msg.ena_mask |= Waypoint::LONG_ENA;
-  if (m_speed_knots_ena) msg.ena_mask |= Waypoint::SPEED_KNOTS_ENA;
-  // note: radius is not passed to autopilot, mission manger will determined if we have arrived at
-  // waypoint then
-  // send autopilot the next waypoint.
-
-  while (0 == waypoint_behavior_pub.getNumSubscribers())
-  {
-    ROS_INFO("Waiting for waypoint subscribers to connect");
-    ros::Duration(0.1).sleep();
-  }
-
-  msg.header.stamp = ros::Time::now();
-
-  waypoint_behavior_pub.publish(msg);
-}
-
-bool WaypointBehavior::checkCorrectedData(const pose_estimator::CorrectedData& data)
-{
-  if (m_behavior_done)
-  {
-    return true;
-  }
-  double dist_to_wp;
-  double currentNorthing;
-  double currentEasting;
-
-  double desiredNorthing;
-  double desiredEasting;
-
-  latLongtoUTM(data.position.latitude, data.position.longitude, &currentNorthing, &currentEasting);
-  latLongtoUTM(m_lat, m_long, &desiredNorthing, &desiredEasting);
-
-  // formula for distance between two 3D points is d=sqrt((x2-x1)^2 + (y2-y1)^2 + (z2-z1)^2)
-  // investigating hypot it looks like it does the squaring and square root to figure out the return
-  // value. Need to add depth or altitude (whatever we are using).
-  //   dist_to_wp = std::hypot((currentNorthing-desiredNorthing, currentEasting-desiredEasting,
-  //   data.depth-m_depth);
-  // dist_to_wp = hypot(abs(currentNorthing-desiredNorthing), abs(currentEasting-desiredEasting));
-
-  dist_to_wp = sqrt(pow((currentNorthing - desiredNorthing), 2) +
-                    pow((currentEasting - desiredEasting), 2) + pow((data.depth - m_depth), 2));
+    dist_to_wp = sqrt(pow((currentNorthing - desiredNorthing), 2) +
+                      pow((currentEasting - desiredEasting), 2) + pow((data.depth - m_depth), 2));
 
-  if (dist_to_wp < m_wp_radius)
-  {
-    m_behavior_done = true;
-    ROS_INFO("We have arrived at waypoint distance to wp [%f] , wp radius [%f]", dist_to_wp,
-             m_wp_radius);
-    return true;
+    if (dist_to_wp < m_wp_radius)
+    {
+      ROS_INFO("We have arrived at waypoint distance to wp [%f] , wp radius [%f]", dist_to_wp,
+               m_wp_radius);
+      behaviorDone = true;
```

Comment:
@patapufate why not setting the action own status instead?

Reply:
You're right. Fixed